src/opd_loader.rs

```diff
@@ -0,0 +1,68 @@
+use std::future::Future;
+
+use crate::PointCloudAsset;
+use bevy::asset::LoadedAsset;
+use bevy::render::render_resource::PrimitiveTopology;
+use bevy::{asset::AssetLoader, prelude::Mesh};
+
+pub struct OpdLoader;
+
+impl OpdLoader {
+    pub fn load_opd<'a>(bytes: &'a [u8]) -> impl Future<Output = PointCloudAsset> + 'a {
+        async move {
+            let file = opd_parser::parse(bytes).unwrap().1;
+            let mut positions = Vec::new();
+
+            let mut max_position = [f32::MIN, f32::MIN, f32::MIN];
+            let mut min_position = [f32::MAX, f32::MAX, f32::MAX];
+
+            for i in file.centroids.into_iter() {
+                let pos: [f32; 3] = i.offset.into();
+                for i in 0..3 {
+                    max_position[i] = max_position[i].max(pos[i]);
+                    min_position[i] = min_position[i].min(pos[i]);
+                }
+                positions.push(pos);
+            }
+
+            let size = [
+                max_position[0] - min_position[0],
+                max_position[1] - min_position[1],
+                max_position[2] - min_position[2],
+            ];
+            for position in positions.iter_mut() {
+                for i in 0..3 {
+                    position[i] = position[i] - min_position[i] - size[i] / 2.0;
```

Comment:
`min_position - size / 2.0` could move outside the loops. Also giving it a descriptive name might make the transformation more obvious.